### PR TITLE
docs: created a page for separating events into new classes.

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR creates a new page in "Advanced Reference" to show people how to separate events into new classes.  

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

File: docpages/advanced_reference/separate-events.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+\page separate-events Separating events into new classes
+
+Now, if you're someone that loves file organisation (or you hate how cluttered your `main.cpp` has become) then you may be interested in moving events into separate classes outside of the `main.cpp` file. This is a great way to improve readability and can be helpful in many cases! For example, you can have two classes on the same event, except one could be reading messages for spam and one could be reading messages for bad words!
+
+In this tutorial, we'll be taking the ["Listening to messages"](/detecting-messages.md) example and moving the `on_message_create` event into a different class.
```
Comment (Commandserver) [inline docpages/advanced_reference/separate-events.md:5]: The link doesn't come up correctly. Use `/ref detecting-messages` instead.

File: docpages/example_programs/interactions_and_components/detecting-messages.md
Comment (Commandserver) [inline docpages/example_programs/interactions_and_components/detecting-messages.md:1]: Should this be changed? The example was okay before. Looks like you accidentally put the class in here.

**Review by Commandserver — CHANGES_REQUESTED** [review]
Looks good overall 👍

File: docpages/example_programs/interactions_and_components/detecting-messages.md
Comment (Jaskowicz1) [reply docpages/example_programs/interactions_and_components/detecting-messages.md:1]: Oh yeah what happened here? This is really weird.

File: docpages/advanced_reference/separate-events.md (hunk 1 above)
Comment (Jaskowicz1) [reply docpages/advanced_reference/separate-events.md:5]: Oh yes you're right, it usually can stay like this except you change .md to .html (look inside detecting-messages.md)

File: docpages/advanced_reference/separate-events.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+\page separate-events Separating events into new classes
+
+Now, if you're someone that loves file organisation (or you hate how cluttered your `main.cpp` has become) then you may be interested in moving events into separate classes outside of the `main.cpp` file. This is a great way to improve readability and can be helpful in many cases! For example, you can have two classes on the same event, except one could be reading messages for spam and one could be reading messages for bad words!
```
Comment (braindigitalis) [inline docpages/advanced_reference/separate-events.md:3]: I would just start the sentence at "If" not "Now, if". It's more direct.

**Review by Commandserver — APPROVED** [review]

File: docpages/advanced_reference/separate-events.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+If you're using CMake, you'll need to add this to your `CMakeLists.txt`. Some IDEs automatically do this but it's always worth double-checking!
+
+Once that's done, it should look similar to this (this screenshot has more files in, so it won't be identical!):
+
+\image html file_example_listeners.png
+
+Now, let's go ahead and write some code in this `message_listener.h`.
+
+~~~~~~~~~~{.cpp}
+#pragma once
```
Comment (braindigitalis) [inline docpages/advanced_reference/separate-events.md:20]: putting non-inline code into a header like this is kind of a no-no.
Can this be redone to have a separate .cpp and .h file where the .h has the forward declaration only?
We would be teaching very bad habits to make them do it this way.

**Review by braindigitalis — CHANGES_REQUESTED** [review]
just a couple of small changes

File: docpages/advanced_reference/separate-events.md (hunk 3 above)
Comment (Jaskowicz1) [reply docpages/advanced_reference/separate-events.md:20]: Yep, we can do this! I didn't realise this was bad practice so that's good to know.

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**